snuba/datasets/table_storage.py

```diff
@@ -18,21 +18,20 @@
 from snuba.snapshots.loaders.single_table import RowProcessor, SingleTableBulkLoader
 from snuba.utils.metrics import MetricsBackend
 from snuba.utils.streams.backends.kafka import KafkaPayload
+from snuba.utils.streams.topics import (
+    Topic,
+    get_topic_creation_config,
+)
 from snuba.writer import BatchWriter
 
-LOG_APPEND_TIME_STORAGES = {
-    StorageKey.EVENTS,
-    StorageKey.ERRORS,
-    StorageKey.TRANSACTIONS,
-}
-
 
 @dataclass(frozen=True)
 class KafkaTopicSpec:
+    topic: Topic
     topic_name: str
```

Comment:
Not sure why we need the topic inside this structure. Shouldn't this be created when resolving a Topic object ?

Reply:
Right, but do we ever need an instance of `KafkaTopicSpec` somewhere from which we take the Topic enum value ? It seems the way we would transform data should be:
`Topic enum value` -> `KafkaTopicSpec instance` -> use the KafkaTopicSpec instance. Now we should not need the enum value anymore. Or am I missing something ?

Reply:
We do. Given a storage that's how we look up the associated topics - we can retrieve the default, commit log or replacement topics this way. Unfortunately our consumers, replacers, subscriptions all need to be able to get hold of the topic enum right now (I think this in part because of the way configuration can be passed via the CLI and needs to be merged with what's provided via settings).

Reply:
Right, storage -> TopicSpec -> Topic. I think we should aim at depending on TopicSpec there where we depend on Topic, but that's beyond the the scope here.